src/gsy_e/models/market/forward.py

```diff
@@ -58,17 +69,50 @@ def _get_start_time(current_time: DateTime) -> DateTime:
     def _get_end_time(current_time: DateTime) -> DateTime:
         """Return time when the market block ends."""
 
+    @property
+    @abstractmethod
+    def _time_from_market_close_till_delivery(self) -> duration:
+        """
+        Retrieves the time duration from the time that the market closes till the time that the
+        traded energy should be delivered.
+        """
+
     @staticmethod
     @abstractmethod
     def _get_market_slot_duration(current_time: DateTime, _config) -> duration:
         """Return duration of market slots inside the market block."""
 
     def create_future_market_slots(self, current_market_time_slot: DateTime,
-                                   config: "SimulationConfig") -> None:
+                                   config: "SimulationConfig") -> List[DateTime]:
         if not ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS:
-            return
-        self._create_future_market_slots(self._get_start_time(current_market_time_slot),
-                                         self._get_end_time(current_market_time_slot), config)
+            return []
+        created_future_slots = self._create_future_market_slots(
+            self._get_start_time(current_market_time_slot),
+            self._get_end_time(current_market_time_slot), config)
+
+        self._set_open_market_slot_parameters(current_market_time_slot, created_future_slots)
+
+        return created_future_slots
+
+    def get_market_parameters_for_market_slot(
+            self, market_slot: DateTime) -> ForwardMarketSlotParameters:
+        """Retrieve the parameters for the selected market slot."""
+        return self._open_market_slot_parameters.get(market_slot)
+
+    def _set_open_market_slot_parameters(
+            self, current_market_slot: DateTime, created_market_slots: List[DateTime]):
+        """Update the parameters of the newly opened market slots."""
+        for market_slot in created_market_slots:
+            if market_slot in self._open_market_slot_parameters:
+                continue
+
+            self._open_market_slot_parameters[market_slot] = ForwardMarketSlotParameters(
+                delivery_start_timestamp=market_slot,
```

Comment:
I'm not sure I understand this. AFAIU, `created_market_slots` contains the slots in which users can post orders for a forward exchange, but these slots aren't the delivery slots of that exchange. So here we're iterating over open slots (not delivery slots) of a forward exchange.
For example, in the case of a `YearForwardMarket` the `_create_future_market_slots()` method will generate something like this:
```py
['2024-01-01T00:00', '2025-01-01T00:00', '2026-01-01T00:00', '2027-01-01T00:00', '2028-01-01T00:00']
```
For each one of these (open) slots, we're now setting:
```py
delivery_start_timestamp=market_slot
```
But this seems odd to me. Shouldn't the delivery start time be the time of the forward exchange slot?
Also: since we're iterating over open slots (not delivery slots), what do the open/close timestamps in the next lines represent?

Maybe I missed some part or I was wrong in one of my assumptions. Let me know if this doesn't make much sense :)

Reply:
I was also confused at first, that's why it took me that long to prepare this draft. The created_market_slots are actually the delivery times. Using your example, the current date is 31-08-2022. This means that the yearly markets that should be open now are the following:
delivery time: 01-01-2024 - > 31-12-2024 . Should be open and close at 31-12-2022. 
delivery time: 01-01-2025 - > 31-12-2025 . Should be open and close at 31-12-2023. 
delivery time: 01-01-2026 - > 31-12-2026 . Should be open and close at 31-12-2024. 
delivery time: 01-01-2027 - > 31-12-2027 . Should be open and close at 31-12-2025. 

Therefore the created_market_slots indeed represent the start of the delivery time.
The confusing part in this is the naming of start / end time IMO. They are not the start and the end time of the market slot, but the time that the rotator should start opening new markets based on a current time, and the time that the rotator should end opening new markets. Hard to name this properly admittedly though. 
One last remark, is that the market for 2028 is opened. This should not be the case since IIRC we only need 4 open yearly markets and not 5. I will also fix it in the context of this PR.

Reply:
> One last remark, is that the market for 2028 is opened. This should not be the case since IIRC we only need 4 open yearly markets and not 5. I will also fix it in the context of this PR.

I'm still a bit confused. From the product document:

> For a yearly product, the market opens 5 years in advance and closes 1 year before 1st day of the year of T0. This means that if you are to sell a product now, you can only sell for 2024-2028 because the yearly period starts on the 1st of the calendar year. The market for 2028 will in this case close on the 1st Jan of 2027 to ensure a yearly gap for the other products to be valid and participate in the market.  For the purposes of visualisation on the UI we will aggregate the data for whole months; the graphs will be yearly viewed with monthly intervals.

From this, it looks like the 2028 slot should be "open" and it will only "close" on the 1st Jan 2027.

I think the terms "market", "slot" and "exchange" create some confusion. If we say that `market == exchange`, then we can say that our market lasts 5 years (this is different from a market slot, which lasts 1 year for Yearly exchanges).

Suppose today is August 2022. 
There must be a gap of 1 year between "now" and the first day of the year/slot when I want to trade. So the exchange can't start in 2023 (less than 1 year from now). The first good start date for the yearly exchange will then be 2024.01.01.

This exchange lasts 5 years, so the slots of this exchange should be `[2024, 2025, 2026, 2027, 2028]` (5 slots).
However, since there must be a gap of 1 year between my order and the first day of the year, the last good moment for me to make an order on the 2028 slot will be the `2027.01.01` (or maybe it should actually be the `2026.12.31`).

Could you please check if this sounds right?

Reply:
> From this, it looks like the 2028 slot should be "open" and it will only "close" on the 1st Jan 2027.

Indeed, I did not recall the product document correctly. Since it is 5 years that will be open and not 4 as I recalled then the implementation is correct. Thanks for checking it out ! 

> I think the terms "market", "slot" and "exchange" create some confusion. If we say that market == exchange, then we can say that our market lasts 5 years (this is different from a market slot, which lasts 1 year for Yearly exchanges).

This is the reason why we use the term "product" more and more these days. I would refrain from using the term exchange, since this is appropriate for gsy-e as a whole and not for individual "markets" / "products". "market" is the term that we use in order to describe traded products that have a specific delivery duration. E.g. the yearly market has a delivery duration of one year. The exact time that the energy will be delivered is determined by the slot, in order to be able to trade for the same product (e.g. yearly) on different timeframes.  

> Suppose today is August 2022.
There must be a gap of 1 year between "now" and the first day of the year/slot when I want to trade. So the exchange can't start in 2023 (less than 1 year from now). The first good start date for the yearly exchange will then be 2024.01.01.

I would urge again to not use the terms start / end, but delivery start / end and open / close in order to not fall into the same pitfall as before. Under these terms, I am rephrasing this:

`There must be a gap of 1 year between "now" and the first day of the year/slot when I want the traded energy to be delivered. So the slot for the delivery year of 2024 can't remain open in 2023 (less than 1 year from now). The slot will be open from now till 31-01-2022, and I can trade till that time if I want energy delivery for 2024.`

Rephrasing the following too:
`Each slot remains open for 5 years, and opens 5+1 years in advance, so the slots of the yearly market should be [2024, 2025, 2026, 2027, 2028] (5 slots).
However, since there must be a gap of 1 year between my order and the first day of the year, the last good moment for me to make an order on the 2028 slot will be the 2026.12.31 end of the day.`